app/models/journeyDomain/routeDetails/loadingAndUnloading/unloading/UnloadingDomain.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.journeyDomain.routeDetails.loadingAndUnloading.unloading
+
+import cats.implicits._
+import models.domain.{GettableAsReaderOps, UserAnswersReader}
+import models.journeyDomain.JourneyDomainModel
+import models.reference.UnLocode
+import pages.routeDetails.loading.PlaceOfLoadingUnLocodePage
+import pages.routeDetails.unloading.AddPlaceOfUnloadingPage
+
+case class UnloadingDomain(
+  unLocode: Option[UnLocode]
+) extends JourneyDomainModel
+
+object UnloadingDomain {
+  // TODO: replace with next page
+
+  implicit val userAnswersReader: UserAnswersReader[UnloadingDomain] = {
+
+    implicit val unLocodeReads: UserAnswersReader[Option[UnLocode]] =
+      AddPlaceOfUnloadingPage.reader.flatMap {
+        case true  => PlaceOfLoadingUnLocodePage.reader.map(Some(_))
+        case false => none[UnLocode].pure[UserAnswersReader]
+      }
```

Comment:
I think this could be refactored using `filterOptionalDependent`

Reply:
Will approve anyway as this file is still a wip

Reply:
Done